secure_message/repository/retriever.py

```diff
@@ -14,6 +14,8 @@
 
 logger = wrap_logger(logging.getLogger(__name__))
 
+THREAD_COUNT_ERROR = "Error retrieving count of threads by survey from database"
```

Comment:
It's inconsistent to have pulled this message out but none of the other repeating ones throughout this PR. I'm personally not a fan of magic strings and would prefer the messages stay in the exceptions for readability purposes, but whichever way you choose could you choose the same one for all of them please :)

Reply:
I've put them all back in with some additional information

Reply:
Can this now be removed?